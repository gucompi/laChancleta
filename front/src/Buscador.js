import React from 'react';
import './App.css';
import {updateBuscador} from './actions/buscadorActions'
import {connect} from 'react-redux'
class Buscador extends React.Component {
  constructor(props){
    super(props);
    this.state={
      buscador:""
    }
  }
  
onChange2 = name => event => {
  let value_ = name === 'image'
  ? event.target.files[0]
  : event.target.value
console.log("this.props")
console.log(this.props)
  this.props.dispatch(updateBuscador(value_))

}
  render(){
    return (
     
      <form className="form-inline search">
      <input onChange={this.onChange2("buscador")} id="boton" className="form-control mr-sm-2" type="search" placeholder="Search" aria-label="Search" />
      <button className="btn btn-outline-success my-2 my-sm-0" type="submit">Search</button>
    </form>
      
    )
  }
}

const mapStateToProps = (state) => {
 
  return {
    buscador:state.buscadorReducer
  }
}
export default connect(mapStateToProps)(Buscador);
