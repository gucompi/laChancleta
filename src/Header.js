import React from 'react';
import './App.css';
import {  Link } from "react-router-dom";

class Header extends React.Component {
  render(){
    return (
     
        <header>
          <nav>
            <nav className="navbar navbar-expand-lg navbar-light bg-light menu">
              <a className="navbar-brand" href="abmstock.html">Amankay</a>
              <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon" />
              </button>
              <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                  <li className="nav-item">
                  <Link to="/form">ABM Producto</Link>
                  </li>
                  <li className="nav-item">
                    <Link to="/stock">Modificar Stock</Link>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link" href="abmstock.html">Facturaci√≥n</a>
                  </li>
                </ul>
              </div>
            </nav>
          </nav>      
        </header>    
      
    )
  }
}

export default Header;
