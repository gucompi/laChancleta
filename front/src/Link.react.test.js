import React from 'react';
import { shallow , mount} from 'enzyme';
import Header from './Header';
import Buscador from './Buscador';
describe('Header', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Header/>);
  
    expect(component).toMatchSnapshot();
  });
});

it('should be possible to activate button with Spacebar', () => {
    const component = mount(<Buscador />);
    component
      .find('input#boton')
      .simulate('keydown', { keyCode: 32 });
    expect(component).toMatchSnapshot();
    component.unmount();
  });