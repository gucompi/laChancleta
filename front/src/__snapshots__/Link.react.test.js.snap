// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`should be possible to activate button with Spacebar 1`] = `ReactWrapper {}`;
